src/compiler/parser.ts

```diff
@@ -1485,6 +1486,11 @@ module ts {
             identifierCount++;
             if (isIdentifier) {
                 let node = <Identifier>createNode(SyntaxKind.Identifier);
+
+                // Set strictModeKind property so that we can report appropriate error later in type checker
+                if (inStrictModeContext() && (token > SyntaxKind.Identifier && token <= SyntaxKind.LastFutureReservedWord)) {

```

Comment:
The check is that if we are in strict mode check if the token is a reserved words or strict mode reserved words. Reserved words and strict-mode reserved words are greater than identifier in SyntaxKind. However if we just check if the token is greater than Identifier, it will include contextual keyword as well.That is why we have to check that token is less than LastFutureReservedWord.